yesod-core/src/Yesod/Routes/TH/RenderRoute.hs

```diff
@@ -182,3 +182,10 @@ notStrict = Bang NoSourceUnpackedness NoSourceStrictness
 
 instanceD :: Cxt -> Type -> [Dec] -> Dec
 instanceD = InstanceD Nothing
+
+conPCompat :: Name -> [Pat] -> Pat
+conPCompat n pats = ConP n
+#if MIN_VERSION_template_haskell(2,18,0)
+                         []
+#endif
+                         pats
```

Comment:
Isn't here an `#else` missing? Or does `ConP` actually have a different number of args in different haskell versions?

Reply:
Yes that's right. There's a new argument in template-haskell 2.18